### ISSUE TITLE
add select tree display refactor

### ISSUE THREAD

**Issue by davidmenendez** [carbon-design-system/ibm-products#2824]
this will probably be the most difficult part of the accessibility refactor, but at Jan's suggestion add select should be refactored to not remove hidden item entries from the dom but instead hidden with css and aria roles. apparently this is the screen reader friendly way of dealing with items.

**Comment by elycheea** [comment]
We should do some more reading on this approach before refactoring for this. I think it’d be helpful for us to understand when this applies since we see this sort of filtering in many cases.

**Comment by davidmenendez** [comment]
after doing additional investigation into this my judgement is that it's not necessary at this time. based on everything i've read and other devs i've consulted with the tradeoffs aren't worth whatever small benefits there may be. my understanding is that with the current implementation and all the additional changes that were made to support additional aria attributes that add select should be in a good place.

i think additional extensive testing with screen reader technology might warrant a follow up, but as it stands i think we can close this issue.
